include/geom/elem.h

```diff
@@ -399,8 +399,13 @@ class Elem : public ReferenceCountedObject<Elem>,
   bool on_boundary () const;
 
   /**
-   * \returns \p true if this element is semilocal to the calling
+   * \returns \p true if this element is "semilocal" to the calling
    * processor, which must specify its rank.
+   *
+   * This method is discouraged, as it uses the *old* definition of
```

Comment:
This is used in MOOSE (for example here: https://github.com/idaholab/moose/blob/devel/framework/include/mesh/MooseMesh.h#L149-L156). There are applications where iterating over all non local point neighbors (and _just_ those) would be super useful.